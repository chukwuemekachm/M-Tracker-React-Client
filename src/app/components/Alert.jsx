import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ message, backgroundColor }) => (
  <div className="ch-alert" style={{ backgroundColor }}>
    <p>
      {message}
    </p>
  </div>
);

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
};

export default Alert;
