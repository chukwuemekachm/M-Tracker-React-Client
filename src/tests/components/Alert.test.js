import React from 'react';
import { shallow } from 'enzyme';
import Alert from '../../app/components/Alert';

describe('Tests Alert component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<Alert message="test" backgroundColor="white" />);
    expect(wrapper.find('div')).toHaveLength(1);
    expect(wrapper.find('p')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
