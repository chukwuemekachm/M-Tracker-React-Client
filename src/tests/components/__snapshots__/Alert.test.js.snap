// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Alert component should render without errors 1`] = `
<div
  className="ch-alert"
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <p>
    test
  </p>
</div>
`;
